Deep selection and shortcuts

kraftdenker opened the pull request:
->Keyboard shortcut on result table to open bookmarks window (CTRL+D)
->Keyboard shortcuts on result table and gallery views to add/remove bookmarks [CTRL+F5 (add) and CHIFT+F5 (remove)] - Hint: Select active bookmark in bookmark window before use.
->Deep mark selection on result table and gallery views using context menu or keyboard shortcuts [CTRL+R (mark highlighted) or SHIFT+R (unmark hightlighted)] - This operations allow mark/unmark all subitems recursively, e.g. mark all attachments from e-mail item or from a compact archive (ZIP), etc...
->Free for use of a Java CAD files viewer for DWG files (CaffViewer).
These features were developed and are been used at SPI/IC/PCDF distro of IPED.

kraftdenker commented:
Need to include the caffviewer.jar into the gitlab repository for copy it to tools directory.

kraftdenker commented:
You are welcome.

Ive not contacted the author (rammi@caff.de) yet, but Ive checked this
issue of viewer's license direct into its site (
https://caff.de/projects/caffviewer/licensing/), where the author
declares that "The de·caff Viewer is freeware, but not Open Source". I
guess that freeware are free to distribute. Initially, ive used the aspose
library (https://products.aspose.com/cad), direct into IPED (not as an
external viewer), but, ive checked that the license didnt permit use it
free of charge, than I cut it out the project.

But you are right about the caff library (not the viewer), that is not
free, as so as the source code.

I will send the author an e-mail to confirm the usage of the viewer.

Em ter., 24 de mar. de 2020 às 17:17, Luis Nassif <notifications@github.com>
escreveu:

> Thank you very much, @kraftdenker <https://github.com/kraftdenker>! I
> will take a look until Friday. Reading caffviewer license, it seems we can
> not distribute the library, just use it. Have you confirmed that with its
> authors? If we can not distribute it, we could enable CADViewer just if
> user have downloaded and put caffviewer.jar into optional_jars dir. What do
> you think?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/lfcnassif/IPED/pull/60#issuecomment-603482473>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AFVAK4WNCOUHATDBRF5YKXTRJEINVANCNFSM4LSYXLSA>
> .
>


-- 
A.M

kraftdenker commented:
His answer, as follow, approves the usage and distribution of caffviewer:

Hi Alberto,

thanks for caring to ask!

Of course you are entitled to use the caffviewer as explained below,
i.e. use and distribute it.


Please note that I'm currently in progress of a major addition to the
viewer so that it can display currently unsupported entities. So you'll
somehow have to care for keeping your distributed viewer up-to-date. The
most stable way to do so is currently to have a look at a file on my
server which the viewer itself uses to check if it needs an update (if
that's explicitly enabled):

        https://caff.de/projects/caffviewer/version.txt

The file's content is a simple as possible, just the following three lines:

# Current version info
VERSION:  3.13.07
MINJVM:   1.8


VERSION should be clear, and MINJVM defines the minimal JVM version
required to run the viewer (here Java 1.8 aka 8).

As this is a bit awkward I just added some command line switches to make
it easier to find out the version of the local viewer, whether there is
an update available, and I even added the possibility to download a
newer version directly from the command line so you can handle this from
scripts.

Use

        java -jar caffviewer.jar -help

to see the possibilities (needs download of the latest version 3.13.07).



Am 24.03.20 um 22:15 schrieb Alberto Magno:
Zitierten Text anzeigen
Hope this helps,
Rammi

--
mailto:rammi@caff.de
http://caff.de/ -- Java stuff

Interested in encrypted communication?
My public PGP Key: http://caff.de/rammi.asc
Alberto Magno <alberto.magno@gmail.com> schrieb am Di., 24. März 2020,
18:02:

> You are welcome.
>
> Ive not contacted the author (rammi@caff.de) yet, but Ive checked this
> issue of viewer's license direct into its site (
> https://caff.de/projects/caffviewer/licensing/), where the author
> declares that "The de·caff Viewer is freeware, but not Open Source". I
> guess that freeware are free to distribute. Initially, ive used the aspose
> library (https://products.aspose.com/cad), direct into IPED (not as an
> external viewer), but, ive checked that the license didnt permit use it
> free of charge, than I cut it out the project.
>
> But you are right about the caff library (not the viewer), that is not
> free, as so as the source code.
>
> I will send the author an e-mail to confirm the usage of the viewer.
>
> Em ter., 24 de mar. de 2020 às 17:17, Luis Nassif <
> notifications@github.com> escreveu:
>
>> Thank you very much, @kraftdenker <https://github.com/kraftdenker>! I
>> will take a look until Friday. Reading caffviewer license, it seems we can
>> not distribute the library, just use it. Have you confirmed that with its
>> authors? If we can not distribute it, we could enable CADViewer just if
>> user have downloaded and put caffviewer.jar into optional_jars dir. What do
>> you think?
>>
>> —
>> You are receiving this because you were mentioned.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/lfcnassif/IPED/pull/60#issuecomment-603482473>, or
>> unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/AFVAK4WNCOUHATDBRF5YKXTRJEINVANCNFSM4LSYXLSA>
>> .
>>
>
>
> --
> A.M
>

lfcnassif commented:
Great news! Thank you, Alberto, for getting in touch!

lfcnassif commented:
Sorry @kraftdenker I was working hard all the week on #67 . I think it is ready now. Will start reviewing this on monday.

kraftdenker commented:
No problem. Take your time.

Luis Nassif <notifications@github.com> schrieb am Fr., 3. Apr. 2020, 20:06:

> Sorry @kraftdenker <https://github.com/kraftdenker> I was working hard
> all the week on #67 <https://github.com/lfcnassif/IPED/pull/67> . I think
> it is ready now. Will start reviewing this on monday.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/lfcnassif/IPED/pull/60#issuecomment-608779994>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AFVAK4U4GIX3XHNX3QBKUXLRKZTW5ANCNFSM4LSYXLSA>
> .
>

lfcnassif commented:
@kraftdenker I finished some (minor and major) fixes and an optimization for deep selection. Also enhanced to let user define keystrokes for each bookmark, just fixed Alt to remove action, I think it is less used. Finally I thought 'shift' modifier a bit dangerous, user could type something somewhere and accidantally trigger actions, so I changed to 'alt' modifier. If you agree, I will merge.

Thanks for the PR.

kraftdenker commented:
"Ready to GO".
I agree with all fixes and optimizations, you may proceed with merge.

I'm sorry about the lack of naming conventions and others issues.  I
intend, in near future, contribute with higher quality code as i will know
IPED better.

  @kraftdenker

Am Do., 9. Apr. 2020 um 12:14 Uhr schrieb Luis Nassif <
notifications@github.com>:

> @kraftdenker <https://github.com/kraftdenker> I finished some (minor and
> major) fixes and an optimization for deep selection. Also enhanced to let
> user define keystrokes for each bookmark, just fixed Alt to remove action,
> I think it is less used. Finally I thought 'shift' modifier a bit
> dangerous, user could type something somewhere and accidantally trigger
> actions, so I changed to 'alt' modifier. If you agree, I will merge.
>
> Thanks for the PR.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/lfcnassif/IPED/pull/60#issuecomment-611583279>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AFVAK4RBV4PPVID3KHMISX3RLXQ4FANCNFSM4LSYXLSA>
> .
>


-- 
A.M

lfcnassif commented:
Just fixed my bookmark shortcut changes in gallery and merged.

Don't worry. IPED has a lot of legacy portuguese code, and its api needs a major refactoring and documentation, my fault.

Thank you for your contribution!
